src/soundsourceproxy.cpp

```diff
@@ -248,51 +193,53 @@ void SoundSourceProxy::loadPlugins() {
         }
     }
 
-    s_soundSourceProviders.finishRegistration();
-
     const QStringList supportedFileExtensions(
-            s_soundSourceProviders.getSupportedFileExtensions());
-    foreach (const QString &supportedFileExtension, supportedFileExtensions) {
-        const Mixxx::SoundSourceProviderPointer pProvider(
-                s_soundSourceProviders.getProviderForFileExtension(supportedFileExtension));
-        const Mixxx::SoundSourcePluginLibraryPointer pPluginLibrary(
-                s_soundSourceProviders.getPluginLibraryForFileExtension(supportedFileExtension));
-        if (pPluginLibrary) {
-            qDebug() << "SoundSourceProvider for" << supportedFileExtension
-                    << "is" << pProvider->getName()
-                    << "@" << pPluginLibrary->getFilePath();
-        } else {
-            qDebug() << "SoundSourceProvider for" << supportedFileExtension
-                    << "is" << pProvider->getName();
+            s_soundSourceProviders.getRegisteredFileExtensions());
+    for (const auto &supportedFileExtension: supportedFileExtensions) {
+        qDebug() << "SoundSource providers for file extension" << supportedFileExtension;
+        const auto registrationsForFileExtension(
+                s_soundSourceProviders.getRegistrationsForFileExtension(
+                        supportedFileExtension));
+        for (const auto& registration: registrationsForFileExtension) {
+            if (registration.getPluginLibrary()) {
+                qDebug() << " " << static_cast<int>(registration.getProviderPriority())
+                        << ":" << registration.getProvider()->getName()
+                        << "@" << registration.getPluginLibrary()->getFilePath();
+            } else {
+                qDebug() << " " << static_cast<int>(registration.getProviderPriority())
+                        << ":" << registration.getProvider()->getName();
+            }
         }
     }
-}
 
-// static
-QStringList SoundSourceProxy::getSupportedFileExtensions() {
-    return s_soundSourceProviders.getSupportedFileExtensions();
+    // Turn the list into a "*.mp3 *.wav *.etc" style string
+    s_supportedFileNamePatterns.clear();
+    for (const auto& supportedFileExtension: supportedFileExtensions) {
+        s_supportedFileNamePatterns += QString("*.%1").arg(supportedFileExtension);
+    }
```

Comment:
And this code is correct, because the individual patterns are concatenated as a QStringList.